nano/core_test/signal_manager.cpp

```diff
@@ -0,0 +1,112 @@
+/**
+ * IMPORTANT NOTE:
+ * These unit tests may or may not work, gtest and boost asio signal handling are not really compatible.
+ * The boost asio signal handling assumes that it is the only one handling signals but gtest
+ * also does some signal handling of its own. In my testing this setup works although in theory
+ * I am playing with unspecified behaviour. If these tests start causing problems then we should
+ * remove them and try some other approach.
+ * The tests are designed as death tests because, as normal tests, the boost library asserts
+ * when I define more than one test case. I have not investigated why, I just turned them into death tests.
+ *
+ * Update: it appears that these tests only work if run in isolation so I am disabling them.
+ */
+
+#include <nano/lib/signal_manager.hpp>
+
+#include <gtest/gtest.h>
+
+#include <boost/format.hpp>
+
+#include <csignal>
+#include <iostream>
+
+static void handler_print_signal (int signum)
+{
+	std::cerr << "boost signal handler " << signum << std::endl
+			  << std::flush;
+}
+
+static int wait_for_sig_received (int millisecs, int & sig_received)
+{
+	for (int i = 0; i < millisecs && sig_received == 0; i++)
+	{
+		usleep (1000);
```

Comment:
This can be replaced with std::this_thread::sleep (std::chrono::microseconds (1));

Reply:
Fixed.